src/extension_instance.ml

```diff
@@ -108,21 +108,68 @@ let stop_server t =
     else Promise.return ()
 ;;
 
-let check_ocaml_lsp_available sandbox =
-  let ocaml_lsp_version sandbox =
-    Sandbox.get_command sandbox "ocamllsp" [ "--version" ]
-  in
-  let cwd =
-    match Workspace.workspaceFolders () with
-    | [ cwd ] -> Some (cwd |> WorkspaceFolder.uri |> Uri.fsPath |> Path.of_string)
-    | _ -> None
+let suggest_to_run_dune_pkg_lock () =
+  let open Promise.Syntax in
+  let (_ : unit Promise.t) =
+    let+ maybe_choice =
+      Window.showWarningMessage
+        ~message:
+          "Dune Package Manager is selected as the active sandbox, but no lock file is \
+           present. Do you want to run dune pkg lock?"
+        ~choices:
+          [ ( "Generate lockfile"
+            , fun () ->
+                let (_ : Ojs.t option Promise.t) =
+                  Vscode.Commands.executeCommand
+                    ~command:Extension_consts.Commands.run_dune_pkg_lock
+                    ~args:[]
+                in
+                () )
+          ; ( "Pick another sandbox"
+            , fun () ->
+                let (_ : Ojs.t option Promise.t) =
+                  Vscode.Commands.executeCommand
+                    ~command:Extension_consts.Commands.select_sandbox
+                    ~args:[]
+                in
+                () )
+          ]
+        ()
+    in
+    Option.iter maybe_choice ~f:(fun f -> f ())
   in
-  Cmd.output ?cwd (ocaml_lsp_version sandbox)
-  |> Promise.Result.fold
-       ~ok:(fun (_ : string) -> ())
-       ~error:(fun (_ : string) ->
-         "Sandbox initialization failed: `ocaml-lsp-server` is not installed in the \
-          current sandbox.")
+  ()
+;;
+
+let check_ocaml_lsp_available t =
```

Comment:
You're changing the parameter from type `sandbox` to type `instance`. However, the only field of `instance` you use is `sandbox`. Why did you make that change?